proto-public/pbmesh/v2beta1/computed_gateway_configuration.proto

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+syntax = "proto3";
+
+package hashicorp.consul.mesh.v2beta1;
+
+import "pbmesh/v2beta1/computed_routes.proto";
+import "pbmesh/v2beta1/pbproxystate/transport_socket.proto";
+import "pbresource/annotations.proto";
+import "pbresource/resource.proto";
+
+// ComputedGatewayConfiguration contains the materialized snapshot of an API
+// Gateway with all of its route and certificate configuration.
+message ComputedGatewayConfiguration {
```

Comment:
This configuration is specific to API gateways; however, the name implies that it applies more broadly to all gateway types. Thoughts on `ComputedAPIGateway` as a prefix for these types to add clarity?

Reply:
yeah I'd agree with @nathancoleman here on the naming, other than that this all looks good to me